src/checkpoint/checkpoint.c

```diff
@@ -841,6 +842,16 @@ finish_write_xids(uint32 chkpnum)
 		LWLockRelease(&oProcData[i].undoStackLocationsFlushLock);
 	}
 
+	/* Check for exited recovery workers with temp files */
+	for (i = 0; i < total_recovery_workers; i++)
+	{
+		if (pg_atomic_unlocked_test_flag(&worker_ptrs[i].has_temp_file))
+		{
+			recovery_load_and_process_state_from_temp_file(i, chkpnum);
+			pg_atomic_clear_flag(&worker_ptrs[i].has_temp_file);
```

Comment:
Why do we clear a flag here?  It means we would be able to use the temp file only once. The idea of this fix is to be able to use this file in any number of checkpoints (we yet know they can be 2).